schema/PMD.yaml

```diff
@@ -0,0 +1,39 @@
+$schema: http://json-schema.org/draft-07/schema#
+$id: https://github.com/lsst-ts/ts_positionmeasurementdevice/blob/master/schema/PositionMeasurementDevice.yaml
+title: PositionMeasurementDevice v1
+description: Schema for PositionMeasurementDevice configuration files
+type: object
+properties:
+  telemetry_interval:
+    description: The interval at which telemetry is published
+    type: number
+    default: 1
+  devices:
+    type: array
+    items:
+      type: object
+      properties:
+        name:
+          type: string
+          default: "Dial Gage"
+      required: [name]
+    minItems: 1
+    maxItems: 8
+  units:
+    type: string
+    default: "um"
+  kind:
+    type: string
+    default: "micrometers"
+  location:
+    type: string
+    default: "Office"
+  serial_port:
+    type: string
+    default: "/dev/ttyUSB0"
+  hub_type:
+    type: string
+    enum: ["Mitutoyo"]
+    default: "Mitutoyo"
+required: [telemetry_interval, devices, units, kind,  location, serial_port, hub_type]
+additionalProperties: false
```

Comment:
I suggest you do something to make sure that a given hub will always be run with a given SAL index. This has a very big advantage to users: a given measurement will always end up in EFD under the same SAL index.

Here are two ways to do this (I'm sure there are others):

1: Most or all of the above configuration is for one hub-specific device. Change the config format to be a collection of sal_index: hub-specific configuration (basically move most or all of the above fields down one level).

This adds some complexity to the schema and makes config files longer, but has two really big advantages:
* There is no danger of running the wrong SAL index with the wrong hub-specific configuration.
* One configuration file is all you need. The CSC uses its SAL index to extract the relevant hub-specific configuration.

I will paste an example next. It's long enough that I'd prefer to keep it separate.

2: Add a required "sal_index" field to your configuration. In the CSC's configure method check this against your actual SAL index and if they do not match then raise an exception.

This makes the configuration format simpler (it's a tiny addition to what you have above) but has some serious disadvantages:
* You need to keep track of which configuration file to run for which SAL index. If you get it wrong the CSC will be unusable.
* You will not learn about your mistake until you send the `start` command to the CSC. By then it's annoyingly late.

(Note: I add a sal_index field in the example for 1, as well, but it serves a different purpose: the CSC looks up the hub-specific configuration using that index).

Reply:
Here is the example.

```
$schema: http://json-schema.org/draft-07/schema#
$id: https://github.com/lsst-ts/ts_positionmeasurementdevice/blob/master/schema/PositionMeasurementDevice.yaml
title: PositionMeasurementDevice v1
description: Schema for PositionMeasurementDevice configuration files
definitions:
  hub_specific_schema:
    type: object
    properties:
      sal_index:
        type: number
      telemetry_interval:
        description: The interval at which telemetry is published
        type: number
        default: 1
      units:
        type: string
        default: "um"
      kind:
        type: string
        default: "micrometers"
      location:
        type: string
        default: "Office"
      serial_port:
        type: string
        default: "/dev/ttyUSB0"
      hub_type:
        type: string
        enum: ["Mitutoyo"]
        default: "Mitutoyo"
    required: [sal_index, telemetry_interval, devices, units, kind,  location, serial_port, hub_type]
    additionalProperties: false
type: object
properties:
  hub_config:
    type: array
    items:
      "$ref": "#/definitions/hub_specific_schema"
required: [hub_config]
additionalProperties: false
```
A configuration might look like:
```
hub_config:
    - sal_index: 5
      location: Office
      ...
    - sal_index: 7:
       location: MTM1M3
       ...
```
though I'm not sure the defaults work in this form. You might have to specify all fields.